test/user.test.js

```diff
@@ -989,6 +989,15 @@ describe('User', function() {
       });
     });
 
+    it('honors unique email for realm', function(done) {
+      User.create(realm1User, function(err, u) {
+        assert(err);
+        assert(err.message.match(/User already exists/) &&
+          err.message.match(/Email already exists/));
+        done();
+      });
+    });
+
```

Comment:
yes it is currently done via the beforeEach test 
https://github.com/strongloop/loopback/blob/eb70a08ca4e34dc3d5f778504b243c5c449902ed/test/user.test.js#L983-L990
where realm1user and realm2user shares same email and username, different password
```js
    var realm1User = {
      realm: 'realm1',
      username: 'foo100',
      email: 'foo100@bar.com',
      password: 'pass100',
    };

    var realm2User = {
      realm: 'realm2',
      username: 'foo100',
      email: 'foo100@bar.com',
      password: 'pass200',
    };
```